jd4/case.py

```diff
@@ -256,20 +257,50 @@ def read_legacy_cases(config, open):
                           memory_bytes,
                           int(score_str))
 
-def read_yaml_cases(config, open):
-    for case in yaml.safe_load(config)['cases']:
-        if 'judge' not in case:
-            yield DefaultCase(partial(open, case['input']),
-                              partial(open, case['output']),
-                              parse_time_ns(case['time']),
-                              parse_memory_bytes(case['memory']),
-                              int(case['score']))
-        else:
-            yield CustomJudgeCase(partial(open, case['input']),
+def read_auto_cases(open, zip_file, time_limit='1s', memory_limit='128m'):
+    prefix = ''
+    cases = []
+    for filename in zip_file.namelist():
+        match = RE_INPUT_FILENAME.match(filename)
+        if match:
+            prefix = match.group(1)
+            cases.append(int(match.group(2)))
+    if not cases:
+        raise FormatError('No test case found.')
+    cases.sort()
+    score = 100 // len(cases)
+    divider = len(cases) - 100 % len(cases)
+    for i, case in enumerate(cases):
+        yield DefaultCase(partial(open, prefix + str(case) + '.in'),
+                          partial(open, prefix + str(case) + '.out'),
+                          parse_time_ns(time_limit),
+                          parse_memory_bytes(memory_limit),
+                          score if i < divider else score+1)
+
+def read_yaml_cases(config, open, zip_file):
+    config = yaml.safe_load(config)
+    if 'cases' not in config:
+        if ('time' not in config) and ('memory' not in config):
```

Comment:
Making both field required means either field not exists, throw a FormatError. Rather than defining an default value.

Reply:
Why we need both field exist?
We can change the time limit , while keeping memory limit as default.

Reply:
Most users won't know the default value in fact. It's common that two fields appear together.

Reply:
OK.